### PR TITLE
Make HealthState show enum 0 value

### PR DESCRIPTION
**Pull request by nikohobart**

### CODE REVIEW

File: pkg/status/status.go [hunk 1]
```diff
@@ -138,3 +145,39 @@ func GetFromStatuses(statuses map[string]*string) (status *common.Status) {
 
 	return
 }
+
+// HealthState requires custom parsing as it is a proto enum. Otherwise enum 0 (NOT_KNOWN) will be an empty string.
+func parseHealth(hstate *common.Health) string {
+	if hstate == nil {
+		return ""
```
Comment (msftpgayam) [inline pkg/status/status.go:152]: Can we return health state as "NOT_KNOWN" if the input hstate is nil?

File: pkg/status/status_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package status
```
Comment (msftpgayam) [inline pkg/status/status_test.go:1]: let's add an unit test for the case where the health state is nil and make sure the reverse conversion is returning NOT_KNOWN.

File: pkg/status/status.go [hunk 3]
```diff
@@ -138,3 +143,39 @@ func GetFromStatuses(statuses map[string]*string) (status *common.Status) {
 
 	return
 }
+
+// HealthState requires custom parsing as it is a proto enum. Otherwise enum 0 (NOT_KNOWN) will be an empty string.
+func parseHealth(hstate *common.Health) string {
+	if hstate == nil {
+		return fmt.Sprintf("currentState:%s", common.HealthState_name[0])
```
Comment (msftpgayam) [inline pkg/status/status.go:150]: Can we use the const `HealthState_NOTKNOWN` instead of 0.

Comment (nikohobart) [reply pkg/status/status.go]: oh good point yes done

**Review by msftpgayam — APPROVED** [review]

**Review by chengzy1994 — APPROVED** [review]
